pandas/tests/groupby/test_categorical.py

```diff
@@ -300,7 +300,7 @@ def test_observed(observed):
     exp_index = CategoricalIndex(
         list("ab"), name="cat", categories=list("abc"), ordered=True
     )
-    expected = DataFrame({"ints": [1.5, 1.5], "val": [20.0, 30]}, index=exp_index)
+    expected = DataFrame({"ints": [1.5, 1.5], "val": [20, 30]}, index=exp_index)
```

Comment:
This was coercing to float with the block code but I don't think that was necessarily desired; fits in an int with new impl

Reply:
Though we might not want this either. Here is code to reproduce:

```python
In [8]:     d = {
   ...:         "ints": [1, 1, 2, 2],
   ...:         "val": [10, 20, 30, 40],
   ...:     }
   ...:     df = pd.DataFrame(d)
In [8]: df.groupby(list("abab")).mean()
Out[7]:
   ints  val
a   1.5   20
b   1.5   30
```

A typical call to `mean` on Series doesn't preserve the int type

Reply:
My intuition is that small changes like this are inevitable when changing from block-wise to column-wise.  Definitely happened with arithmetic changeover.